code/game/dna/genes/vg_powers.dm

```diff
@@ -46,82 +46,119 @@
 
 	var/new_eyes = input("Please select eye color.", "Character Generation", rgb(M.r_eyes,M.g_eyes,M.b_eyes)) as null|color
 	if(new_eyes)
-		M.r_eyes = hex2num(copytext(new_eyes, 2, 4))
-		M.g_eyes = hex2num(copytext(new_eyes, 4, 6))
-		M.b_eyes = hex2num(copytext(new_eyes, 6, 8))
```

Comment:
The color you pick is stored in `new_eyes`, but then `new_eyes` isn't used for anything. If something actually changes your eye color, it isn't this.